packages/data-provider/src/services/dictionaryService.ts

```diff
@@ -1,36 +1,57 @@
-import { Dictionary } from '@overture-stack/lyric-data-model';
+import { Category, Dictionary, NewCategory, NewDictionary } from '@overture-stack/lyric-data-model';
 import { BaseDependencies } from '../config/config.js';
 import categoryRepository from '../repository/categoryRepository.js';
-import getCategoryUtils from '../utils/categoryUtils.js';
 import getDictionaryUtils from '../utils/dictionaryUtils.js';
 
 const dictionaryService = (dependencies: BaseDependencies) => {
 	const LOG_MODULE = 'DICTIONARY_SERVICE';
 	const { logger } = dependencies;
 	return {
-		register: async (categoryName: string, dictionaryName: string, version: string): Promise<Dictionary> => {
+		register: async (
+			categoryName: string,
+			dictionaryName: string,
+			version: string,
+		): Promise<{ dictionary: Dictionary; category: Category }> => {
 			logger.debug(
 				LOG_MODULE,
 				`Register new dictionary categoryName '${categoryName}' dictionaryName '${dictionaryName}' version '${version}'`,
 			);
-			const { createCategoryIfDoesNotExist } = getCategoryUtils(dependencies);
-			const categoryRepo = categoryRepository(dependencies);
-
-			const savedCategory = await createCategoryIfDoesNotExist(categoryName);
 
+			const categoryRepo = categoryRepository(dependencies);
 			const { createDictionaryIfDoesNotExist, fetchDictionaryByVersion } = getDictionaryUtils(dependencies);
+
 			const dictionary = await fetchDictionaryByVersion(dictionaryName, version);
 
-			const savedDictionary = await createDictionaryIfDoesNotExist(
-				dictionaryName,
-				version,
-				savedCategory,
-				dictionary.schemas,
-			);
+			const savedDictionary = await createDictionaryIfDoesNotExist(dictionaryName, version, dictionary.schemas);
+
+			// Check if Category exist
+			const foundCategory = await categoryRepo.getCategoryByName(categoryName);
+
+			if (!foundCategory) {
```

Comment:
Will suggest using the opposite check (`if (foundCategory)`) for the following reasons:
- negative logic checks are slightly trickier to understand when reading
- there is a lot of function handling when the function is not found, resulting in a lot of nested code
- the positive check returns immediately

So I would replace this with:
```ts
		if (foundCategory) {
			logger.info(LOG_MODULE, `Dictionary and Category already exists`);
			return { dictionary: savedDictionary, category: foundCategory };
		}
		// continue with code for case where foundCategory === false
```

Reply:
this function handles 3 scenarios:

- foundCategory && foundCategory.activeDictionaryId === savedDictionary.id
- foundCategory && foundCategory.activeDictionaryId !== savedDictionary.id
- !foundCategory

moving the positive check first would look like this
```
if (foundCategory && foundCategory.activeDictionaryId === savedDictionary.id) {
  // code here
} else if(foundCategory && foundCategory.activeDictionaryId !== savedDictionary.id) {
   // code here
} else{ 
   // code here
} 
```